tor/add_onion.go

```diff
@@ -70,7 +91,33 @@ func (f *OnionFile) PrivateKey(_ OnionType) ([]byte, error) {
 	if _, err := os.Stat(f.privateKeyPath); os.IsNotExist(err) {
 		return nil, ErrNoPrivateKey
 	}
-	return ioutil.ReadFile(f.privateKeyPath)
+	privateKey, err := ioutil.ReadFile(f.privateKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the privateKey doesn't start with either v2 or v3 key params
+	// it's likely encrypted.
+	if !bytes.HasPrefix(privateKey, []byte(v2keyParam)) &&
+		!bytes.HasPrefix(privateKey, []byte(v3keyParam)) {
+		// If the privateKey is encrypted but --tor.encryptkey
+		// wasn't set we log an error and stop
+		if !f.encryptKey {
+			return nil, fmt.Errorf(
+				"It appears the Tor private key is " +
+					"encrypted but you didn't pass the --tor.encryptkey flag. " +
+					"Please restart lnd with the --tor.encryptkey flag or delete " +
+					"the Tor key file for regeneration.\n",
+			)
```

Comment:
Somewhat of a library nit: this error message is specific to lnd, but the `tor` package may be used by other applications. We should return a concrete error here (e.g. `ErrEncryptedPrivateKey`) and inspect it at a higher level within lnd (the caller of `AddOnion`) to return the user-friendly error.

Reply:
Ah, yeah that makes sense. I like that better. I'll update

Reply:
@wpaulino updated!